chessington/engine/pieces.py

```diff
@@ -52,37 +102,89 @@ def get_available_moves(self, board):
         return []
 
 
+class MoveByDirection:
+    FIRST_STEP = 1
+
+    @staticmethod
+    def in_bounds(move) -> bool:
+        return move.row in range(0, 8) and move.col in range(0, 8)
+
+    @staticmethod
+    def get_position(current_square, current_piece, board, list_moves, step, direction):
+        next_square = Square.at(current_square.row + direction.row * step,
+                                current_square.col + direction.col * step)
+        if MoveByDirection.in_bounds(next_square):
+            piece = board.get_piece(next_square)
+            if not piece:
+                list_moves.append(next_square)
+            else:
+                if piece.player != current_piece.player:
+                    list_moves.append(next_square)
+                return 0
+        else:
+            return 0
+        return 1
+
+    @staticmethod
+    def get_moves(current_piece, board, directions, not_king):
+        current_square = board.find_piece(current_piece)
+        list_moves = []
+        for direction in directions:
+            continue_moves = MoveByDirection.get_position(current_square, current_piece, board, list_moves,
+                                                          MoveByDirection.FIRST_STEP, direction)
+            if not_king and continue_moves:
+                step = 2
+                while True:
+                    continue_moves = MoveByDirection.get_position(current_square, current_piece, board,
+                                                                  list_moves, step, direction)
+                    if not continue_moves:
+                        break
+                    step += 1
+
+        return list_moves
```

Comment:
I am a bit confused of why is this a class? In theory, classes _are_ things and functions _do_ things. Looking at the behaviour of all the methods here it looks like all them *do* stuff.